### PR TITLE
Update to Cats 1.0.0-RC1, removing iteratee-fs2 module

### PR DESCRIPTION
**Pull request by travisbrown**
~~This is broken because of "`monadError.flatMap` from `tailRecM` consistency" failures I haven't looked at yet.~~ (Should be fixed by #219.)

I've removed the iteratee-fs2 module now that fs2 no longer provides its own `Task`. We'll need to replace it with a module for cats-effect, which we may also want to use in iteratee-files (I've got a branch working on this). In the meantime, though, I'm planning to publish 0.15.0 as soon as the test failures here are resolved.

### CODE REVIEW

File: testing/shared/src/main/scala/io/iteratee/testing/EqInstances.scala [hunk 1]
```diff
@@ -9,8 +9,8 @@ import scala.Predef._
 trait EqInstances {
   implicit def eqThrowable: Eq[Throwable] = Eq.fromUniversalEquals
 
-  implicit def eqEnumerator[F[_]: Monad, A: Eq](implicit eq: Eq[F[Vector[A]]]): Eq[Enumerator[F, A]] =
-    eq.on[Enumerator[F, A]](_.toVector)
+  implicit def eqEnumerator[F[_]: Monad, A: Eq](implicit eqFVA: Eq[F[Vector[A]]]): Eq[Enumerator[F, A]] =
+    Eq.by[Enumerator[F, A], F[Vector[A]]](_.toVector)(eqFVA)
```
Comment (rkrzewski) [inline testing/shared/src/main/scala/io/iteratee/testing/EqInstances.scala]: scalac does good job with inference here
```scala
Eq.by(_.toVector)
```
is sufficient

Comment (travisbrown) [reply testing/shared/src/main/scala/io/iteratee/testing/EqInstances.scala]: Oh, thanks—I'd tried that but with the explicit `eqFVA`, which doesn't work. I guess there's no reason to pass the `F[Vector[A]]` instance explicitly, though.

**Comment by ImLiar** [comment]
@travisbrown It seems that the bug could be related to `arbitraryVectorIteratee` implicit. Returning constant iteratee instead of `oneOf` "solves" the problem. Does it sound like this arbitrary resolved/called twice, returning different iteratees to compare their output later?

**Comment by travisbrown** [comment]
@ImLiar Yes, that sounds likely. I just tried narrowing this instance down, and this subset passes:
```scala

        it <- Gen.oneOf[Iteratee[F, Vector[A], Vector[A]]](
          //Iteratee.done[F, Vector[A], Vector[A]](as),
          Iteratee.drop[F, Vector[A]](n).flatMap(_ => F),
          //Iteratee.drop[F, Vector[A]](n).flatMap(_ => M.pure(as)),
          Iteratee.head[F, Vector[A]].map(_.getOrElse(Vector.empty)),
          Iteratee.peek[F, Vector[A]].flatMap(_ => F),
          Iteratee.peek[F, Vector[A]].flatMap(head =>
            M.pure(as ++ head.fold(Vector.empty[A])(_.take(n)))
          ),
          //Iteratee.take[F, Vector[A]](n).flatMap(taken => M.pure(taken.flatMap(_.headOption))),
          Iteratee.identity[F, Vector[A]].flatMap(_ => F)//,
          //Iteratee.identity[F, Vector[A]].flatMap(_ => M.pure(as))
        )
```
Haven't investigated further, though.

**Comment by travisbrown** [comment]
For anyone following along, it turns out that this is a real bug in `tailRecM`—e.g. here's something close to a minimization:

```scala
import cats.Id, cats.implicits._, io.iteratee._

val F = Iteratee.iterateeMonad[Id, Int]

val done = Iteratee.done[Id, Int, Int](1)
val whatever = Iteratee.fold[Id, Int, Int](0)(_ + _)

val fm = done.flatMap(_ => whatever)

val tr = F.tailRecM[Option[Int], Int](None) {
  case None => F.map(done)(v => Left(Some(v)))
  case Some(_) => F.map(whatever)(Right(_))
}

val values = Enumerator.enumerate[Id, Int](1, 2, 3)

val good = values.into(fm) // good: cats.Id[Int] = 6
val bad = values.into(tr) // bad: cats.Id[Int] = 0
```

The issue is that we don't check whether the initial `f(a)` result is done before feeding it. I'll have a quick fix in today but in the long run we should think about nicer ways to avoid losing data by accidentally feeding completed iteratees.

**Pull request merged**